src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.AutoRedirect.cs

```diff
@@ -45,6 +45,10 @@ public static IEnumerable<object[]> RedirectStatusCodesOldMethodsNewMethods()
             {
                 yield return new object[] { statusCode, "GET", "GET" };
                 yield return new object[] { statusCode, "POST", statusCode <= 303 ? "GET" : "POST" };
+                yield return new object[] { statusCode, "PUT", statusCode == 303 ? "GET" : "PUT" };
+                yield return new object[] { statusCode, "DELETE", statusCode == 303 ? "GET" : "DELETE" };
+                yield return new object[] { statusCode, "PATCH", statusCode == 303 ? "GET" : "PATCH" };
+                yield return new object[] { statusCode, "OPTIONS", statusCode == 303 ? "GET" : "OPTIONS" };
```

Comment:
The one thing I'd add here is a completely custom method, e.g. "MYCUSTOMMETHOD" or whatever, and validate it gets changed to GET on 303.